### DISCUSSION TITLE
Account registration and roster management

### DISCUSSION THREAD

**Discussion by allen90132** [vitalyster/SharpXMPP#121] [Ideas]
Hi all,

I'm new for Xamarin chat app.  I find a function in this project. How to use XmppRegisterHandler.cs to register a new account ? Also need add/remove roster, thanks.

**Comment by vitalyster** [comment]
This library was written long time ago and it is not ready to use for full-featured chat application and nothing is documented.
You need to have strong XMPP and XEP knowledge to be able extend this library for your needs.
1. New account registration is described in XEP-0077 - `XmppRegisterHandler` only have basic XEP-0077 payload parser, you also need another `XmppConnection` subclass to process registration payloads in correct flow.
This is quite hard task for beginner, however, most of XMPP servers provide web registration and you may not need "inband" account registration - just open registration page in your app's WebView.
2. Roster management is described in [XMPP-IM RFC](https://xmpp.org/rfcs/rfc6121.html#roster-add), it is not implemented there right now, but you can easily implement it looking at `RosterManager.cs` - it have `get` query implemented, while to add new user into roster you just need `set` request.
